packages/esm-patient-list-app/src/PatientTable/patient-table.component.tsx

```diff
@@ -0,0 +1,140 @@
+import { navigate } from '@openmrs/esm-framework';
+import { Link } from 'carbon-components-react';
+import DataTable, {
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from 'carbon-components-react/lib/components/DataTable';
+import DataTableSkeleton from 'carbon-components-react/lib/components/DataTableSkeleton';
+import Pagination from 'carbon-components-react/lib/components/Pagination';
+import Search from 'carbon-components-react/lib/components/Search';
+import { debounce } from 'lodash';
+import React, { useMemo, CSSProperties } from 'react';
+import styles from './patient-table.component.scss';
+
+const PatientTable: React.FC<PatientTableProps> = ({ patients, columns, search, pagination, isLoading, autoFocus }) => {
+  const rows: Array<any> = useMemo(
+    () =>
+      patients.map((patient) => {
+        const row = {};
+        columns.forEach((column) => {
+          row['id'] = patient['id'] || patient['uuid'];
+          const value = column.getValue ? column.getValue(patient) : patient[column.key];
+          row[column.key] = column.link ? (
+            <Link
+              onClick={(e) => {
+                e.preventDefault();
+                navigate({ to: column.link.getUrl(patient) });
+              }}>
+              {value}
+            </Link>
+          ) : (
+            value
+          );
+        });
+        return row;
+      }),
+    [patients, columns],
+  );
+  const handleSearch = React.useMemo(() => debounce((searchTerm) => search.onSearch(searchTerm), 300), []);
+
+  return (
+    <>
+      {!isLoading ? (
```

Comment:
You could use this notation:
```tsx
if(isLoading)
  return <Skeleton />

return <Table />
```

Not a big difference, but make it easier to read